### PR TITLE
feat: Add AI features unit and e2e tests

### PR DESCRIPTION
**Pull request by alexgrozav**
- Adds unit tests for @n8n/chat package
  image

- Adds unit tests for various editor-ui files
- Adds e2e tests for AI Nodes and new input types
- Refactor editor-ui for unit testing
  - Pinia triggers a strange error when stores are imported from an index file, this PR fixes it
  ```
   FAIL  src/components/__tests__/WorkflowSettings.spec.ts [ src/components/__tests__/WorkflowSettings.spec.ts ]
  TypeError: Cannot read properties of undefined (reading '$id')
   ❯ ../../node_modules/.pnpm/pinia@2.1.6_typescript@5.2.2_vue@3.3.4/node_modules/pinia/dist/pinia.mjs:1803:26
      1801|     return stores.reduce((reduced, useStore) => {
      1802|         // @ts-expect-error: $id is added by defineStore
      1803|         reduced[useStore.$id + mapStoreSuffix] = function () {
         |                          ^
      1804|             return useStore(this.$pinia);
      1805|         };
   ❯ Module.mapStores 
  ../../node_modules/.pnpm/pinia@2.1.6_typescript@5.2.2_vue@3.3.4/node_modules/pinia/dist/pinia.mjs:1801:19
   ❯ src/components/WorkflowSettings.vue:454:6
   ❯ src/components/__tests__/WorkflowSettings.spec.ts:1:31
  ```

### CODE REVIEW

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]
LOVE IT! 💙 Just remove the console.logs and its good to go in my view

File: packages/editor-ui/src/mixins/pushConnection.ts [hunk 1]
```diff
@@ -281,6 +281,9 @@ export const pushConnection = defineComponent({
 			}
 
 			if (receivedData.type === 'executionFinished' || receivedData.type === 'executionRecovered') {
+				console.log('IT HAS ARRIVED');
+				console.log(receivedData, this.workflowsStore.activeExecutionId);
+
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/editor-ui/src/mixins/pushConnection.ts]: Good catch!

File: cypress/utils/executions.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			mode: 'manual',
+			startedAt: new Date().toISOString(),
+			stoppedAt: new Date().toISOString(),
+			status: 'success',
+			finished: true,
+		},
+	};
+}
+
+export function runMockWorkflowExcution({
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply cypress/utils/executions.ts:78]: Thank you!  🙏

File: packages/editor-ui/src/__tests__/defaults.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
+import type { INodeTypeData, INodeTypeDescription } from 'n8n-workflow';
+import {
+	AGENT_NODE_TYPE,
+	MANUAL_CHAT_TRIGGER_NODE_TYPE,
+	MANUAL_TRIGGER_NODE_TYPE,
+} from '@/constants';
+import { NodeConnectionType } from 'n8n-workflow';
+
+export const testingNodeTypes: INodeTypeData = {
+	[MANUAL_TRIGGER_NODE_TYPE]: {
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/editor-ui/src/__tests__/defaults.ts:17]: Updated to use `nodes.json`. Great idea!

**Review by OlegIvaniv — APPROVED** [review]

**Pull request merged**